test/postControllerTest.js

```diff
@@ -0,0 +1,187 @@
+/* global it, describe */
+const express = require('express');
+const request = require('supertest');
+const postController = require('../controllers/postController');
+const sinon = require('sinon');
+const postService = require('../services/postService');
+const assert = require('assert');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(express.static('../public'));
+
+app.use('', postController);
+
+describe('postController', () => {
+  let stubGetAllPosts = sinon.stub();
+  let stubGetOnePost = sinon.stub();
+  let stubCreateNewPost = sinon.stub();
+
+  beforeEach(() => {
+    stubGetAllPosts = sinon.stub(postService, 'getAllPosts').resolves('expected');
+    stubGetOnePost = sinon.stub(postService, 'getOnePost').resolves('expected');
+    stubCreateNewPost = sinon.stub(postService, 'createNewPost').returns(Promise.resolve('expected'));
+  });
+
+  afterEach(() => {
+    stubGetAllPosts.restore();
+    stubGetOnePost.restore();
+    stubCreateNewPost.restore();
+  });
+
+  describe('GET /', () => {
+    it('should call method "postService.getAllPosts" with no arguments', (done) => {
+      request(app)
+        .get('/')
+        .end((err) => {
+          sinon.assert.calledWith(stubGetAllPosts);
+          if (err) return done(err);
+          return done();
+        });
+    });
+  });
+  describe('GET /', () => {
+    it('should return correct "posts" in promise', (done) => {
+      request(app)
+        .get('/')
+        .end((err) => {
+          stubGetAllPosts().then((posts) => {
+            assert.equal(posts, 'expected');
```

Comment:
This is testing whether a sinon stub works, and doesn't test anything in postController

Since this is testing Sinon, Sinon should already have a test like this: https://github.com/sinonjs/sinon/blob/988df0c5253368182d60d1141d09586e6e00f39f/test/stub-test.js#L161

Reply:
How can I test if my controller is passing ```posts``` to a view when it handles ```GET /```?

Since I am mocking out postService and it is in the promise resolution of postService.getAllPosts that my posts is being passed to res.render.
So the only way seems not mock out postService. Then would I leave this testing to my future e2e tests?

Reply:
Typically in a unit test, you'd be looking at what arguments are passed to `res.render`.

However, since you are writing an integration test, then you should probably be looking at the html that is rendered (ie: after the view template is rendered with the arguments).

If this sounds weird to you, it's because it is... an integration at this level tests not only your controller, but also express and your views.

I do think that you do need tests for your views, but they don't neccessarily belong here... But since you are going to convert your views to react in the future, we can put them here for the time being